### PR TITLE
story/VETHUB-69

### PR DESCRIPTION
**Pull request by freshc0w**
Compartmentalised the checkbox and labels to its own component and encapsulated the logic of manipulating state and `onChange` event away from each other. 

Functionality also added for the selection of course, subject and system.

```bash
git fetch
git checkout story/VETHUB-69
```

**Testing:**
1. Run `yarn dev` and try checking the checkbox.
2. Run `yarn test` and all test cases should pass.

### CODE REVIEW

File: src/components/MCQ/MCQTagOptions.vue [hunk 1]
```diff
@@ -1,29 +1,43 @@
...
+  system: [],
+});
+
+const modifySelectedTags = (
+  isChecked: boolean,
+  { category, topic }: { category: keyof SelectedTags; topic: string },
+): void => {
+  selectedTags.value[category] = isChecked
+    ? [...selectedTags.value[category], topic]
+    : selectedTags.value[category].filter((t) => t !== topic);
```
Comment (anzau) [inline src/components/MCQ/MCQTagOptions.vue]: Avoid singular letters on Higher Order Functions

File: src/components/FilterCheckbox.vue [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        :value="topic"
+        @change="onChecked($event)"
+      />
+      <label :for="`${category}-${topic}-checkbox`">{{ topic }}</label>
+    </li>
+  </ul>
+</template>
+
+<script setup lang="ts">
+import { defineProps } from "vue";
```
Comment (anzau) [inline src/components/FilterCheckbox.vue]: Think you don't need this, there's a warning when you run `yarn dev`

**Review by anzau — APPROVED** [review]
Just a couple of comments but all good for me.

**Review by Mohamad11Dab — APPROVED** [review]
Apart of what Anzah mentioned, LGTM !

**Pull request merged**